app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -99,4 +110,85 @@ class MainActivity : AppCompatActivity() {
         }
         dialog.show()
     }
+
+    private fun editLayerDialog(oldLayerName: String) {
+        viewModel.isEditDialogOpen = true
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(this).apply {
+            setTitle(context.getString(R.string.editLayerDialogTitle))
+            setView(view)
+            val errorView = view.findViewById<TextView>(R.id.errorText)
+            val layerNameInput = view.findViewById<EditText>(R.id.layerNameField)
+            while (!viewModel.defaultLayerHintFound) {
+                val currentHint = getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+                viewModel.createLayerHint(currentHint)
+            }
+            layerNameInput.hint = getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+            if (viewModel.isEditDialogOpen && viewModel.newName != null) {
+                layerNameInput.setText(viewModel.newName)
+                errorView.isVisible = layerNameInput.text.toString().length > NAME_LENGTH_MAX
```

Comment:
should this be a trimmed string?

Reply:
This resets the already trimmed name acquired from the user editing into the text field, it shouldn't be possible for it to have extra whitespace for this check